src/braket/aws/aws_device.py

```diff
@@ -399,31 +401,34 @@ def get_devices(
         device_map = {}
         device_regions_set = AwsDevice._get_devices_regions_set(arns, provider_names)
         for region in device_regions_set:
-            session_for_region = AwsDevice._copy_aws_session(aws_session, [region])
-            # Require simulators to be instantiated
-            # in the same region as the AWS session
-            region_device_types = sorted(
-                types
-                if region == aws_session.boto_session.region_name
-                else types - {AwsDeviceType.SIMULATOR}
-            )
-            region_device_arns = [
-                result["deviceArn"]
-                for result in session_for_region.search_devices(
-                    arns=arns,
-                    names=names,
-                    types=region_device_types,
-                    statuses=statuses,
-                    provider_names=provider_names,
+            session_region = aws_session.boto_session.region_name
+            # If the region is not the same as the current session's region
+            # and only simulators are requested, then search_devices will not
+            # be called at all because simulators are only retrieved for the
+            # current region
+            if region == session_region or types != {AwsDeviceType.SIMULATOR}:
+                session_for_region = AwsDevice._copy_aws_session(aws_session, [region])
+                # Simulators are only instantiated in the same region as the AWS session
+                region_device_types = sorted(
+                    types if region == session_region else types - {AwsDeviceType.SIMULATOR}
+                )
```

Comment:
Is this now redundant?

Reply:
Nope, because it's still possible to ask for both QPUs and simulators, in which case on QPUs will be searched